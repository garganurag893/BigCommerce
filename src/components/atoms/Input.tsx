import { InputHTMLAttributes, FC } from "react";

interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
  label?: string;
  helperText?: string;
  grid?: 1 | 2 | 3;
  error?: string;
}

const Input: FC<InputProps> = ({
  label,
  helperText,
  grid = 1,
  error,
  ...props
}) => {
  let gridWidth = "";
  if (grid === 2) {
    gridWidth = "md:w-1/2";
  } else if (grid === 3) {
    gridWidth = "md:w-1/3";
  }
  return (
    <div className={`w-full px-3 mb-4 md:mb-0 ${gridWidth}`}>
      {label && (
        <label className="block uppercase tracking-wide text-gray-700 text-xs font-bold mb-2">
          {label}
        </label>
      )}
      <input
        className={`appearance-none block w-full bg-gray-200 text-gray-700 rounded py-3 px-4 mb-3 leading-tight focus:outline-none focus:bg-white ${
          !!error ? "border border-red-500" : ""
        }`}
        {...props}
      />
      {helperText && (
        <p className="text-gray-600 text-xs italic">{helperText}</p>
      )}
      {!!error && <p className="text-red-500 text-xs italic">{error}</p>}
    </div>
  );
};

export default Input;
