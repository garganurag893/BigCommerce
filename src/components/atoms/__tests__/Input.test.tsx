import { render, fireEvent } from '@testing-library/react'

import Input from '../Input'

const sampleText = 'First Name'
const sampleClassName = 'testClass'
const sampleHelperText = 'Some random name'

describe('Test Input Component', () => {
  it('Check for Input tag in DOM', () => {
    const onClickHandler = jest.fn()
    const { getByRole, getByText } = render(
      <Input
        label={sampleText}
        helperText={sampleHelperText}
        error={sampleHelperText}
        className={sampleClassName}
        onClick={onClickHandler}
        grid={1}
      />,
    )

    expect(getByRole('textbox')).toBeInTheDocument()
    expect(getByText(sampleText)).toBeInTheDocument()
    expect(getByRole('textbox')).toHaveClass(sampleClassName)

    fireEvent.click(getByRole('textbox'))

    expect(onClickHandler).toHaveBeenCalledTimes(1)
  })

  it('Input Snapshot Test', () => {
    const { container } = render(
      <Input
        label={sampleText}
        helperText={sampleHelperText}
        className={sampleClassName}
      />,
    )
    expect(container).toMatchSnapshot()
  })

  it('Input Grid 2 Snapshot Test', () => {
    const { container } = render(
      <Input
        label={sampleText}
        helperText={sampleHelperText}
        className={sampleClassName}
        grid={2}
      />,
    )
    expect(container).toMatchSnapshot()
  })

  it('Input Grid 3 Snapshot Test', () => {
    const { container } = render(
      <Input
        label={sampleText}
        helperText={sampleHelperText}
        className={sampleClassName}
        grid={3}
      />,
    )
    expect(container).toMatchSnapshot()
  })

  it('Input With Error Snapshot Test', () => {
    const { container } = render(
      <Input
        label={sampleText}
        error={sampleHelperText}
        className={sampleClassName}
        grid={3}
      />,
    )
    expect(container).toMatchSnapshot()
  })
})
