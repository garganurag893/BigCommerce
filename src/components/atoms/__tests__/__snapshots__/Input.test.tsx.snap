// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Input Component Input Grid 2 Snapshot Test 1`] = `
<div>
  <div
    class="w-full px-3 mb-4 md:mb-0 md:w-1/2"
  >
    <label
      class="block uppercase tracking-wide text-gray-700 text-xs font-bold mb-2"
    >
      First Name
    </label>
    <input
      class="testClass"
    />
    <p
      class="text-gray-600 text-xs italic"
    >
      Some random name
    </p>
  </div>
</div>
`;

exports[`Test Input Component Input Grid 3 Snapshot Test 1`] = `
<div>
  <div
    class="w-full px-3 mb-4 md:mb-0 md:w-1/3"
  >
    <label
      class="block uppercase tracking-wide text-gray-700 text-xs font-bold mb-2"
    >
      First Name
    </label>
    <input
      class="testClass"
    />
    <p
      class="text-gray-600 text-xs italic"
    >
      Some random name
    </p>
  </div>
</div>
`;

exports[`Test Input Component Input Snapshot Test 1`] = `
<div>
  <div
    class="w-full px-3 mb-4 md:mb-0 "
  >
    <label
      class="block uppercase tracking-wide text-gray-700 text-xs font-bold mb-2"
    >
      First Name
    </label>
    <input
      class="testClass"
    />
    <p
      class="text-gray-600 text-xs italic"
    >
      Some random name
    </p>
  </div>
</div>
`;

exports[`Test Input Component Input With Error Snapshot Test 1`] = `
<div>
  <div
    class="w-full px-3 mb-4 md:mb-0 md:w-1/3"
  >
    <label
      class="block uppercase tracking-wide text-gray-700 text-xs font-bold mb-2"
    >
      First Name
    </label>
    <input
      class="testClass"
    />
    <p
      class="text-red-500 text-xs italic"
    >
      Some random name
    </p>
  </div>
</div>
`;
